### PR TITLE
Populate Mach-O relocs from chained fixups and parse and use chained imports

### PR DESCRIPTION
**Pull request by thestr4ng3r**
# DO NOT SQUASH

**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [x] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [x] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

For details see the commit descriptions.

Summary:
* Relocs are created from all chained fixups (`ir`)
* New chained imports are parsed too and replace the old info from indirect syms (`ii`)
* Combining the two points above, relocs also have their name set to the symbol name (displayed in `ir`)
* The above point also has the effect that imported objc superclasses are resolved correctly (`ic`, see updated tests)
* Chained bind fixups are now also patched to have meaningful pointers to imports rather than the unpatched chained fixup value (looks like garbage data otherwise)

**Test plan**

Before (relocs are incomplete, `objc_alloc(0x8020000000000005)` makes no sense):
Bildschirmfoto 2023-06-24 um 16 10 06

After:
Bildschirmfoto 2023-06-24 um 16 09 18

### CODE REVIEW

File: librz/bin/format/mach0/mach0_chained_fixups.c [hunk 1]
```diff
@@ -0,0 +1,573 @@
...
+RZ_API size_t MACH0_(chained_imports_count)(struct MACH0_(obj_t) * obj) {
+	return rz_vector_len(&obj->chained_fixups.imports);
+}
+
+/**
+ * Copy contents of a dyld_chained_import* of the binary into \p dst
+ * \param ordinal index of the import, given by chained pointers
+ * \return true if an import with the given ordinal was found and read successfully
+ */
+RZ_API RZ_OWN bool MACH0_(get_chained_import)(struct MACH0_(obj_t) * obj, ut32 ordinal, struct MACH0_(chained_import_t) * dst) {
```
Comment (XVilka) [inline librz/bin/format/mach0/mach0_chained_fixups.c]: Unnecessary marker for a `bool`

File: librz/bin/format/mach0/mach0_relocs.c [hunk 2]
```diff
@@ -148,384 +149,393 @@ static int parse_import_ptr(struct MACH0_(obj_t) * bin, struct reloc_t *reloc, i
...
+}
+
+static void parse_relocs_from_extrels(RzSkipList *dst, struct MACH0_(obj_t) * bin) {
+	if (!bin->symtab || !bin->dysymtab.extreloff || !bin->dysymtab.nextrel) {
+		return;
+	}
+	parse_relocation_info(bin, dst, bin->dysymtab.extreloff, bin->dysymtab.nextrel);
+}
+
+RZ_BORROW RzSkipList *MACH0_(get_relocs)(struct MACH0_(obj_t) * bin) {
```
Comment (XVilka) [inline librz/bin/format/mach0/mach0_relocs.c]: Add list element type in comments please

**Review by XVilka — CHANGES_REQUESTED** [review]
A tremendous amount of work! Their logic is so complex...
LGTM with two minor things.

**Review by XVilka — CHANGES_REQUESTED** [review]
```
Run python3 rz-bindgen/src/lint.py \
Type comment at <rizin/librz/bin/format/mach0/mach0.h:21[7](https://github.com/rizinorg/rizin/actions/runs/5366343464/jobs/9735715767?pr=3602#step:6:8):37> for unknown type RzSkipList
Type comment at <rizin/librz/bin/format/mach0/mach0.h:244:46> for unknown type RzSkipList
Type comment at <rizin/librz/bin/format/mach0/mach0_relocs.c:521:46> for unknown type RzSkipList
```

**Review by wargio — APPROVED** [review]

**Comment by XVilka** [comment]
@thestr4ng3r does it also resolve https://github.com/rizinorg/rizin/issues/3459?

**Comment by thestr4ng3r** [comment]
I don't think so, seems unrelated.

**Review by XVilka — APPROVED** [review]

**Pull request merged**